inference-engine/tests/functional/shared_test_classes/src/single_layer/detection_output.cpp

```diff
@@ -90,7 +90,13 @@ void DetectionOutputLayerTest::GenerateInputs() {
     }
 }
 
-void DetectionOutputLayerTest::Compare(const std::vector<std::uint8_t> &expected, const InferenceEngine::Blob::Ptr &actual) {
```

Comment:
May be we should make the DetectionOutput comparator to be smarter and ignore difference in order of elements with the same probability?

Also, how about updating the specification?

Also, what is the behaviour of current implementation in plugins?

Reply:
> May be we should make the DetectionOutput comparator to be smarter and ignore difference in order of elements with the same probability?

Yeah, modifying the comparator seems like a less invasive idea, allows for more flexibility in the implementations too. I'll get right on it.
> Also, how about updating the specification?

Could you elaborate on that?
> Also, what is the behaviour of current implementation in plugins?

They either use stable sort or compare indices in the comparator if the scores are equal (which makes the sorting de facto stable).

Reply:
> > May be we should make the DetectionOutput comparator to be smarter and ignore difference in order of elements with the same probability?

> Yeah, modifying the comparator seems like a less invasive idea, allows for more flexibility in the implementations too. I'll get right on it.

Actually, it's not that simple. I just remembered why I settled for the invasive way - the comparator overhaul would work in most cases but not all. That's because some elements may be discarded right after sorting (because of the top_k/keep_top_k params). That's a problem because the cutoff might be in between two elements seen as equal by the sorting algorithm. Without a rigid comparator rule, different elems can be cut off depending on the implementation.

Reply:
Agree that the case mentioned above could happen. But it is really rare and in 99.99% of cases the smart comparator would work.

Regarding the specification. We should update the `docs/ops/detection/DetectionOutput_1.md` with the note about sorting elements with equal probability.

Reply:
@lazarevevgeny Does this change need to be in this MR?

Reply:
The change in the comparator function could be in another PR. But update for the documentation should be done in this one.

Reply:
Updated the documentation.